asserts/crypto.go

```diff
@@ -28,26 +28,77 @@ import (
 	"encoding/base64"
 	"encoding/hex"
 	"fmt"
+	"io"
 	"time"
 
 	"golang.org/x/crypto/openpgp/packet"
 )
 
-// TODO: eventually this should be the only non-test file using/importing directly from golang.org/x/crypto
+const (
+	maxEncodeLineLength = 76
+)
 
-func generatePrivateKey() (*packet.PrivateKey, error) {
-	priv, err := rsa.GenerateKey(rand.Reader, 2048)
+func encodeFormatAndData(format string, data []byte) []byte {
+	buf := bytes.NewBufferString(format)
+	buf.WriteByte(' ')
+	buf.Grow(base64.StdEncoding.EncodedLen(len(data)))
+	enc := base64.NewEncoder(base64.StdEncoding, buf)
+	enc.Write(data)
+	enc.Close()
+	flat := buf.Bytes()
+
+	buf = new(bytes.Buffer)
+	flatSize := len(flat)
+	lines := flatSize / maxEncodeLineLength
+	rest := flatSize % maxEncodeLineLength
+	newlines := lines
+	if rest > 0 {
+		newlines++
+	}
+
+	buf.Grow(flatSize + newlines)
+	off := 0
+	for i := 0; i < lines; i++ {
+		buf.Write(flat[off : off+maxEncodeLineLength])
+		buf.WriteByte('\n')
+		off += maxEncodeLineLength
+	}
+	if rest > 0 {
+		buf.Write(flat[off:])
+		buf.WriteByte('\n')
+	}
+
+	return buf.Bytes()
+}
+
+type openpgpSerializer interface {
+	openpgpSerialize(w io.Writer) error
+}
+
+func encodeOpenpgp(obj interface{}, kind string) ([]byte, error) {
+	serializer, ok := obj.(openpgpSerializer)
```

Comment:
@niemeyer because otherwise the check would need to be repeated in the callers, and we need the check until we split PublicKey along two different interfaces, which I think we need for clarity, but I want to do that after the demo is out